hadoop-connector-examples/src/main/java/io/pravega/example/hadoop/PravegaSingleSegmentOutputFormat.java

```diff
@@ -30,41 +28,45 @@
 import java.util.Optional;
 
 /**
- * A special PravegaOutputFormat to write events to Pravega in global order, which means:
- *   For each reducer (or maper if no reducer) the job creates a separate Pravega stream;
- *   There is only one segment for the stream;
- *   Events are ordered inside a stream;
- *   Events are globally orders across all the output streams.
+ * A special PravegaOutputFormat to write events to Pravega stream in sequence:
+ *  each mapper/reducer task will write to a different stream, with the name as "OUTPUT_STREAM_PREFIX" + taskId;
+ *  each stream is with a fixed(1) segment, so the order of event is same as order of writing.
+ *
+ * In the end, the outcome streams are also "in order" which means:
+ *  all the keys of events inside a stream are in order (here the key is the first 10 bytes of event);
+ *  any key of stream with lower taskId is smaller than all the keys of streams with greater taskId.
  *
- * (This is to mimic how Hadoop Terasort generates one file per reducer after sorting)
  */
-public class PravegaGlobalOrderOutputFormat<V> extends PravegaOutputFormat<V> {
+public class PravegaSingleSegmentOutputFormat<V> extends PravegaFixedSegmentsOutputFormat<V> {
```

Comment:
`PravegaSingleSegmentOutputFormat` is a special case of `PravegaFixedSegmentsOutputFormat` in which we have that `PRAVEGA_NUMBER_OF_SEGMENTS` is `1`. I suggest we remove `PravegaSingleSegmentOutputFormat` and use `PravegaFixedSegmentsOutputFormat` with `PRAVEGA_NUMBER_OF_SEGMENTS = 1`.

If that doesn't work, then I'd like to understand why.

Reply:
PravegaSingleSegmentOutputFormat extends PravegaFixedSegmentsOutputFormat but also does one special thing - every reducer will write to a different stream (each stream is of single segment). This is to keep the sequence of terasort result.
I've just renamed PravegaSingleSegmentOutputFormat to PravegaTeraSortOutputFormat to avoid confusion, hope it clarifies.